extensions/panache/panache-rest-common/deployment/src/main/java/io/quarkus/panache/rest/common/deployment/methods/ListHalMethodImplementor.java

```diff
@@ -50,11 +80,47 @@ public void implement(ClassCreator classCreator) {
         ResourceAnnotator.addPath(methodCreator, UrlImplementor.getCollectionUrl(entityClassName));
         ResourceAnnotator.addProduces(methodCreator, ResourceAnnotator.APPLICATION_HAL_JSON);
 
-        ResultHandle entities = dataAccessImplementor.listAll(methodCreator);
-        methodCreator.returnValue(wrapEntities(methodCreator, entities, entityClassName));
+        ResultHandle queryParameters = getQueryParameters(methodCreator);
+        BranchResult ifHasLimit = methodCreator.ifTrue(hasLimit(methodCreator, queryParameters));
+
+        // Has limit
+        AssignableResultHandle limit = ifHasLimit.trueBranch().createVariable(int.class);
```

Comment:
Very minor: I think it would be a little easier to read if `ifHasLimit.trueBranch()` was given a well named variable